### PR TITLE
Dev 11 auth

### PR DESCRIPTION
**Pull request by finlaylp**
opens home if already logged in, login screen if not; most auth functionality done (logging in, swapping from signup to login); no email confirmation on signup yet; no forgot password;

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -107,40 +112,20 @@ export default function App() {
     console.log("Firebase setup already complete.")
   }
 
+  const user = firebase.auth().currentUser;
+  var startupScreen = 'Login'
```
Comment (rohithdara) [inline App.js:143]: just curious why this is a var instead of const

File: App.js [hunk 2]
```diff
@@ -107,40 +112,20 @@ export default function App() {
...
-          <Tab.Screen name="Calendar" component={Calendar} />
-          <Tab.Screen name="Profile" component={Profile} />
-        </Tab.Navigator>
+        <Stack.Navigator initialRouteName={startupScreen}>
+          <Stack.Screen name="Login" component={login} options={{headerShown:false}}/>
+          <Stack.Screen name="Signup" component={signup} options={{headerShown:false}}/>
+          <Stack.Screen name="Root" component={Root} options={{headerShown:false}} />
+        </Stack.Navigator>
       </NavigationContainer>
       </PaperProvider>
```
Comment (rohithdara) [inline App.js:157]: tab inconsistency I think (like </PaperProvider> should be 1 tab back)

File: components/authentication/login.jsx [hunk 3]
```diff
@@ -1,163 +1,144 @@
...
-export default class login extends React.Component {
-    state={
-        email:"",
-        password:"",
-        showPassword:true,
-    }
+export default ({ navigation }) => {
+  const [email, setEmail] = React.useState('');
+  const [password, setPassword] = React.useState('');
+  const [showPassword, setShowPassword] = React.useState(true);
```
Comment (rohithdara) [inline components/authentication/login.jsx]: Should the default for showpassword be true?

**Review by rohithdara — COMMENTED** [review]
Looks good to me my guy. Would be helpful if you included screenshots of what the screens themselves look like

File: App.js (hunk 1 above)
Comment (finlaylp) [reply App.js:143]: because it isn't constant, it gets reassigned a couple lines down

File: components/authentication/login.jsx (hunk 3 above)
Comment (finlaylp) [reply components/authentication/login.jsx]: I guess I should change it, but true means that the password is hidden.

Comment (rohithdara) [reply components/authentication/login.jsx]: oh then true makes sense, I assumed the opposite

Comment (rohithdara) [reply components/authentication/login.jsx]: but yeah I guess variable name wise that is confusing

Comment (finlaylp) [reply components/authentication/login.jsx]: yeah I'll reverse

**Review by jpoist97 — COMMENTED** [review]
Looks good, I'm trusting you've already tested all of the login/signup stuff cause I can't figure out how to log out and do it again hahaha. Good shit tho. Only thing is that when I'm typing in my password on my iphone I can't actually see the textbox im typing in since the keyboard covers it. I used KeyboardAwareFlatList for this in the LogWorkout component and im sure there's something similar that you can use for this

File: components/authentication/signup.jsx [hunk 4]
```diff
@@ -1,169 +1,154 @@
...
-                            >
-                    </Image>
-                    <TextInput style={styles.nameInput}
-                                selectionColor={"#5DB075"}
-                                placeholder={'Name'}
-                                mode={'flat'}
-                                underlineColor={'transparent'}
-                                >
+          default:
+            console.log(errorMessage);
```
Comment (ctperry0301) [inline components/authentication/signup.jsx:37]: Add a way to display the error message to the user.

File: components/authentication/signup.jsx [hunk 5]
```diff
@@ -1,169 +1,154 @@
...
+    <SafeAreaView style={{ flex: 1 }}>
+      <View style={styles.container}>
+        <Image
+          style={styles.icon}
+          source={require('../../assets/icon.png')}
+        />
+        <TextInput
+          style={styles.input}
+          selectionColor="#A192FF"
+          placeholder="Name"
```
Comment (ctperry0301) [inline components/authentication/signup.jsx]: Maybe change placeholder to first name, cause that's all we care about

File: components/authentication/login.jsx [hunk 6]
```diff
@@ -1,163 +1,144 @@
...
+        }
+      });
+  }
 
-    signUpPressed = () => {
-        alert("sign up pressed")
-    }
+  function signupPress() {
+    navigation.navigate('Signup');
+    console.log("Navigating to signup...")
```
Comment (ctperry0301) [inline components/authentication/login.jsx:33]: Display message to user if they have an incorrect password.

**Review by ctperry0301 — DISMISSED** [review]
Looks really good. After fixing the comments, should be good to go ahead and merge.

File: components/authentication/login.jsx (hunk 6 above)
Comment (finlaylp) [reply components/authentication/login.jsx:33]: This part is just if they don't have an account yet and navigate to the signup page from the login page, so no info is saved

**Review by ctperry0301 — APPROVED** [review]
Looks good to me

**Pull request merged**